Stop serializing commit record for read-only transactions

swimj opened the pull request:
Added an IsReadOnly function to CommitRecord to indicate if the transaction generating that commit record was read-only or not.  When the logger is processing it's flush queue, if it encounters a CommitRecord, it uses this function to determine whether it should skip serialization. For read-only transactions, there is no need to write out the commit record to anywhere, since no modifications were made.  In that case, the commit record is only needed for invoking the transaction's callback function.

fix #209.

tli2 commented on test/storage/log_test.cpp:
```diff
@@ -47,8 +47,10 @@ class WriteAheadLoggingTests : public TerrierTest {
+      // Okay to fill in null since nobody will invoke the callback.
+      // is_read_only argument is set to false, because we do not write out a commit record for a transaction if it is
+      // not read-only.
+      return storage::CommitRecord::Initialize(buf, txn_begin, txn_commit, nullptr, nullptr, false);
```
Can you add a test to verify that read-only transactions are not logged out?

swimj commented:
Adjusted LargeLogTest to have a 50-50 update-select ratio (previously was all selects) and properly account for read-only transactions with this feature implemented.

Added ReadOnlyTransactionsGenerateNoLogTest to test that no log records are generated by read only transactions, which is the primary goal of this PR.

tli2 commented:
You need to fix documentation for something according to Travis
